tests/stdlib/trst.nim

```diff
@@ -428,3 +428,46 @@ suite "RST inline markup":
           rnLeaf  'lnk'
           rnLeaf  '___'
       """)
+
+  test "no punctuation in the end of a standalone URI is allowed":
+    check(dedent"""
+        [see (http://no.org)], end""".toAst ==
+      dedent"""
+        rnInner
+          rnLeaf  '['
+          rnLeaf  'see'
+          rnLeaf  ' '
+          rnLeaf  '('
+          rnStandaloneHyperlink
+            rnLeaf  'http'
```

Comment:
not blocking this PR but why isn't the whole url a single leaf node?

Reply:
pre-existing way:-)
agree, changed to single